examples/src/test/java/jdocs/state/DurableStateChangeEventDocExample.java

```diff
@@ -0,0 +1,198 @@
+        return items.isEmpty();
+      }
+
+      public boolean hasItem(String itemId) {
+        return items.containsKey(itemId);
+      }
+
+      public State updateItem(String itemId, int quantity) {
+        if (quantity == 0) {
+          items.remove(itemId);
+        } else {
+          items.put(itemId, quantity);
+        }
+        return this;
+      }
+
+      public State removeItem(String itemId) {
+        items.remove(itemId);
+        return this;
+      }
+
+    }
+
+    public interface Command extends CborSerializable {
+    }
+
+    public static class AddItem implements Command {
+      public final String itemId;
+      public final int quantity;
+      public final ActorRef<Done> replyTo;
+
+      public AddItem(String itemId, int quantity, ActorRef<Done> replyTo) {
+        this.itemId = itemId;
+        this.quantity = quantity;
+        this.replyTo = replyTo;
+      }
+    }
+
+    public static class RemoveItem implements Command {
+      public final String itemId;
+      public final ActorRef<Done> replyTo;
+
+      public RemoveItem(String itemId, ActorRef<Done> replyTo) {
+        this.itemId = itemId;
+        this.replyTo = replyTo;
+      }
+    }
+
+    public static class DiscardCart implements Command {
+      public final ActorRef<Done> replyTo;
+
+      @JsonCreator
+      public DiscardCart(ActorRef<Done> replyTo) {
+        this.replyTo = replyTo;
+      }
+    }
+
+    public interface ChangeEvent extends CborSerializable {
+    }
+
+    public static final class ItemAdded implements ChangeEvent {
+      public final String itemId;
+      public final int quantity;
+
+      public ItemAdded(String itemId, int quantity) {
+        this.itemId = itemId;
+        this.quantity = quantity;
+      }
+    }
+
+    public static final class ItemRemoved implements ChangeEvent {
+      public final String itemId;
+
+      @JsonCreator
+      public ItemRemoved(String itemId) {
+        this.itemId = itemId;
+      }
+    }
+
+    public static final class CartRemoved implements ChangeEvent {
+    }
+
+    public static EntityTypeKey<Command> ENTITY_TYPE_KEY =
+        EntityTypeKey.create(Command.class, "ShoppingCart");
+
+    public static Behavior<Command> create(String cartId) {
+      return new ShoppingCart(cartId);
+    }
+
+    private final String cartId;
+
+    private ShoppingCart(String cartId) {
+      super(
+          PersistenceId.of(ENTITY_TYPE_KEY.name(), cartId),
+          SupervisorStrategy.restartWithBackoff(Duration.ofMillis(200), Duration.ofSeconds(5), 0.1));
+      this.cartId = cartId;
+    }
+
+    @Override
+    public State emptyState() {
+      return new State();
+    }
+
+    // #changeEventHandler
+
+    @Override
+    public ChangeEventHandler<Command, State, ChangeEvent> changeEventHandler() {
+      return new ChangeEventHandler<Command, State, ChangeEvent>() {
+        @Override
+        public ChangeEvent changeEvent(State previousState, State newState, Command command) {
```

Comment:
I renamed the methods in https://github.com/akka/akka/pull/32254/commits/9fe13135505846ca0ec74a8b11eddcdd7515f7cb because this was somewhat unclear that it is for update vs delete.